tests/lava/proc/sparse/test_model.py

```diff
@@ -0,0 +1,1452 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+import unittest
+from lava.magma.core.learning.learning_rule import Loihi2FLearningRule
+from scipy.sparse import csr_matrix
+
+from lava.proc.sparse.process import Sparse
+from lava.proc.dense.process import Dense, LearningDense
+from lava.proc.sparse.process import Sparse, DelaySparse, LearningSparse
+from lava.proc.learning_rules.stdp_learning_rule import STDPLoihi
+from lava.proc.dense.process import Dense
+from lava.proc.sparse.process import Sparse, DelaySparse
+from lava.proc.sparse.models import AbstractPyDelaySparseModel as APDSM
+from lava.proc.io.source import RingBuffer as Source
+from lava.proc.io.sink import RingBuffer as Sink
+import unittest
+from lava.magma.core.run_configs import Loihi2SimCfg
+import numpy as np
+from lava.magma.core.decorator import implements, requires, tag
+from lava.magma.core.model.py.model import PyLoihiProcessModel
+from lava.magma.core.model.py.ports import PyOutPort, PyInPort
+from lava.magma.core.model.py.type import LavaPyType
+from lava.magma.core.process.ports.ports import OutPort, InPort
+from lava.magma.core.process.process import AbstractProcess
+from lava.magma.core.process.variable import Var
+from lava.magma.core.resources import CPU
+from lava.magma.core.run_conditions import RunSteps
+from lava.magma.core.sync.protocols.loihi_protocol import LoihiProtocol
+from lava.proc.dense.models import AbstractPyDelayDenseModel
+from lava.utils.weightutils import SignMode
+from tests.lava.magma.core import learning
+
+
+def create_network(input_data, conn, weights):
+    source = Source(data=input_data)
+    sink = Sink(shape=(weights.shape[0], ),
+                buffer=input_data.shape[1])
+
+    source.s_out.connect(conn.s_in)
+    conn.a_out.connect(sink.a_in)
+
+    return source, conn, sink
+
+
+def create_learning_network(data_pre, conn, data_post, weights=None):
+    pre = Source(data=data_pre)
+    post = Source(data=data_post)
+
+    if weights is not None:
+        sink = Sink(shape=(weights.shape[0], ),
+                    buffer=data_post.shape[1])
+
+        conn.a_out.connect(sink.a_in)
+
+    pre.s_out.connect(conn.s_in)
+    post.s_out.connect(conn.s_in_bap)
+
+    if weights is not None:
+        return pre, conn, post, sink
+    else:
+        return pre, conn, post
+
+
+class TestSparseProcessModelFloat(unittest.TestCase):
+    """Tests for Sparse class in floating point precision. """
+
+    # TODO add dedicated tests for get and set
+    def test_consistency_with_dense_random_shape(self):
+        """Tests if the results of Sparse and Dense are consistent. """
+
+        simtime = 10
+        shape = np.random.randint(1, 300, 2).tolist()
+        weights = (np.random.random(shape) - 0.5) * 2
+
+        # sparsify
```

Comment:
Comments should start with capital letter (whole file)

Reply:
done